src/extension/commands/client_init.c

```diff
@@ -32,6 +36,53 @@ static const dd_command_spec _spec = {
     .config_features_cb = dd_command_process_config_features_unexpected,
 };
 
+typedef struct url {
+    char *host;
+    unsigned int port;
+} url;
+
+void free_url(url *out)
+{
+    if (out && out->host) {
+        free(out->host);
+        out->host = NULL;
+    }
+}
+
+void extract_url(url *out)
```

Comment:
I think you misunderstood, I meant that you should just parse and add to msgpack here not to recreate everything with custom types. For example, something like this: https://github.com/DataDog/dd-appsec-php/blob/274e64f2b0c99ef57496ce978d784f489b069e8f/src/extension/commands/request_init.c#L248